Instrument Index Management with user action telemetry

cjcenizal opened the pull request:
Actions tracked:

* "Load app" action
* "Clear cache" action (both individual and in bulk)
* "Delete index" action (both individual and in bulk)
* "Flush index" action (both individual and in bulk)
* "Force merge index" action (both individual and in bulk)
* "Close index" action (both individual and in bulk)
* "Open index" action (both individual and in bulk)
* "Freeze index" action (both individual and in bulk)
* "Unfreeze index" action (both individual and in bulk)
* "Refresh index" action (both individual and in bulk)
* "Update settings" action (in details panel)
* "Open details panel" action
* "View summary tab" action
* "View mappings tab" action
* "View stats tab" action
* "View settings tab" action
* "View edit settings tab" action

Telemetry shape as reported by `http://localhost:5601/api/stats?extended=true`:

```
{  
  "user_actions":[  
    {  
      "key":"app_load",
      "value":1
    },
    {  
      "key":"index_clear_cache",
      "value":1
    },
    {  
      "key":"index_clear_cache_many",
      "value":1
    },
    {  
      "key":"index_close",
      "value":1
    },
    {  
      "key":"index_close_many",
      "value":1
    },
    {  
      "key":"index_delete",
      "value":1
    },
    {  
      "key":"index_delete_many",
      "value":1
    },
    {  
      "key":"index_flush",
      "value":1
    },
    {  
      "key":"index_flush_many",
      "value":1
    },
    {  
      "key":"index_force_merge",
      "value":1
    },
    {  
      "key":"index_force_merge_many",
      "value":1
    },
    {  
      "key":"index_freeze",
      "value":1
    },
    {  
      "key":"index_freeze_many",
      "value":1
    },
    {  
      "key":"index_open",
      "value":2
    },
    {  
      "key":"index_open_many",
      "value":1
    },
    {  
      "key":"index_refresh",
      "value":1
    },
    {  
      "key":"index_refresh_many",
      "value":1
    },
    {  
      "key":"index_unfreeze",
      "value":1
    },
    {  
      "key":"index_unfreeze_many",
      "value":1
    },
    {  
      "key":"show_details_click",
      "value":3
    },
    {  
      "key":"detail_panel_edit_settings_tab",
      "value":3
    },
    {  
      "key":"detail_panel_mapping_tab",
      "value":1
    },
    {  
      "key":"detail_panel_settings_tab",
      "value":2
    },
    {  
      "key":"detail_panel_stats_tab",
      "value":1
    },
    {  
      "key":"detail_panel_summary_tab",
      "value":1
    }
  ]
}
```

elasticmachine commented:
Pinging @elastic/es-ui

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/3968/)

cjcenizal commented:
@bmcconaghy I've updated the PR per our discussion by removing the `trackUserRequest` helper and moving this logic into `api.js`. I've also added tracking for when the app is first loaded by the user, and for the "refresh index" action which I missed the first time around. Could you please take another look?

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/4644/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/4658/)

cjcenizal commented:
Retest

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/4672/)

sebelga commented on x-pack/plugins/index_management/public/app.js:
```diff
@@ -4,17 +4,26 @@
-  </div>
-);
+export class App extends Component {
+  componentWillMount() {
```
This lifecycle is deprecated (https://reactjs.org/blog/2018/03/29/react-v-16-3.html). We should use `componentDidMount()`

sebelga commented on x-pack/plugins/index_management/public/services/api.js:
```diff
@@ -32,6 +66,9 @@ export async function closeIndices(indices) {
     indices
   };
   const response = await httpClient.post(`${apiPrefix}/indices/close`, body);
```
For consistency, I think it would have been better:

```js
const actionType = indices.length > 1 ? UA_INDEX_CLOSE_MANY : UA_INDEX_CLOSE;
const request = httpClient.post(`${apiPrefix}/indices/close`, body);
const response = await trackUserRequest(request, actionType);
...
```

You know, I'm all about consistency 😊 Aside, the `trackUserRequest()` (that will come as a public method from the userAction plugin in the new platform) being an abstraction, we could easily extend it's behaviour in in a single place whenever needed.

cjcenizal replied:
We discussed this, found some ways to make `trackUserRequest` easier to understand, and agreed it's OK to have different implementations from app-to-app for now, with an eye towards unifying them in the future once the New Platform is in a state that supports this effort.
